runtime/wasm/src/host_exports.rs

```diff
@@ -812,6 +813,19 @@ pub(crate) fn bytes_to_string(logger: &Logger, bytes: Vec<u8>) -> String {
     s.trim_end_matches('\u{0000}').to_string()
 }
 
+pub(crate) fn abi_encode(params: Vec<Token>) -> Result<Vec<u8>, anyhow::Error> {
+    Ok(encode(&params))
+}
+
+pub(crate) fn abi_decode(types: String, data: Vec<u8>) -> Result<Token, anyhow::Error> {
+    let param_types =
+        Reader::read(&types).or_else(|e| Err(anyhow::anyhow!("Failed to read types: {}", e)))?;
+
+    decode(&[param_types], &data)
+        .map(|mut tokens| tokens.pop().unwrap())
```

Comment:
Please add a comment explaining why it's ok to `.pop().unwrap()` and discard the vec.

Reply:
Fixed!